tests/e2e/etcd_grpcproxy_test.go

```diff
@@ -0,0 +1,180 @@
+
+	proxyCtl := e2e.NewEtcdctl(&e2e.EtcdProcessClusterConfig{}, []string{proxyClientURL})
+	err = proxyCtl.Put("k1", "v1", config.PutOptions{})
+	require.NoError(t, err)
+
+	memberCtl := e2e.NewEtcdctl(&e2e.EtcdProcessClusterConfig{}, []string{node1ClientURL})
+	_, err = memberCtl.MemberAdd(node2Name, []string{node2PeerURL})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Run new member
+	proc2, err := runEtcdNode(
+		node2Name, t.TempDir(),
+		node2ClientURL, node2PeerURL,
+		"existing", fmt.Sprintf("%s=%s,%s=%s", node1Name, node1PeerURL, node2Name, node2PeerURL),
+	)
+	require.NoError(t, err)
+
+	// Wait for auto sync of endpoints
+	err = waitForEndpointInLog(proxyProc, node2ClientURL)
+	require.NoError(t, err)
+
+	memberList, err := memberCtl.MemberList()
+	require.NoError(t, err)
+
+	node1MemberID, err := findMemberIDByEndpoint(memberList.Members, node1ClientURL)
+	require.NoError(t, err)
+
+	// Second node could be not ready yet
+	for i := 0; i < 10; i++ {
+		_, err = memberCtl.MemberRemove(node1MemberID)
+		if err != nil && strings.Contains(err.Error(), rpctypes.ErrGRPCUnhealthy.Error()) {
+			time.Sleep(500 * time.Millisecond)
+			continue
+		}
+		break
+	}
+
+	// Remove node1 from member list and stop this nod
+	require.NoError(t, err)
+	require.NoError(t, proc1.Stop())
+
+	var resp *clientv3.GetResponse
+	for i := 0; i < 10; i++ {
+		resp, err = proxyCtl.Get("k1", config.GetOptions{})
+		if err != nil && strings.Contains(err.Error(), rpctypes.ErrGRPCLeaderChanged.Error()) {
+			time.Sleep(500 * time.Millisecond)
+			continue
+		}
+	}
+	require.NoError(t, err)
+	kvs := testutils.KeyValuesFromGetResponse(resp)
+	assert.Equal(t, []testutils.KV{{Key: "k1", Val: "v1"}}, kvs)
+
+	require.NoError(t, proc2.Stop())
+	require.NoError(t, proxyProc.Stop())
+}
+
+func runEtcdNode(name, dataDir, clientURL, peerURL, clusterState, initialCluster string) (*expect.ExpectProcess, error) {
+	proc, err := e2e.SpawnCmd([]string{e2e.BinDir + "/etcd",
+		"--name", name,
+		"--data-dir", dataDir,
+		"--listen-client-urls", clientURL, "--advertise-client-urls", clientURL,
+		"--listen-peer-urls", peerURL, "--initial-advertise-peer-urls", peerURL,
+		"--initial-cluster-token", "etcd-cluster",
+		"--initial-cluster-state", clusterState,
+		"--initial-cluster", initialCluster,
+	}, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = proc.Expect("ready to serve client requests")
+
+	return proc, err
+}
+
+func findMemberIDByEndpoint(members []*etcdserverpb.Member, endpoint string) (uint64, error) {
+	for _, m := range members {
+		if m.ClientURLs[0] == endpoint {
+			return m.ID, nil
+		}
+	}
+
+	return 0, fmt.Errorf("member not found")
+}
+
+func waitForEndpointInLog(proxyProc *expect.ExpectProcess, endpoint string) error {
+	endpoint = strings.Replace(endpoint, "http://", "", 1)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	// ExpectFunc reads existing lines, but we're expecting for the events from the future,
```

Comment:
Maybe I'm missing something, but I think this statement is not true. `ExpectFunc` has an infinite for loop that only ends if we match the line or there is an error reported (expected when program exit). So it should be able to read from future events, if it doesn't this is a bug.

Reply:
Yes, you're right. My mind somehow kept old interpretation of what this function does. Yesterday I changed my mind, but today forget it once again, lol.

Get rid of redundant code.